### ISSUE TITLE
readFile behaviour not aligned with fs module

### ISSUE THREAD

**Issue by demetriusnunes** [RiptideElements/s3fs#150]
I detected 2 issues when using readFile that is different from the "fs" module counterpart:
* encoding cannot be passed in as a string in the second parameter
* `data.Body.toString(encoding)` is only called when using the non-promise version of the method  causes inconsistencies in its usage: https://github.com/RiptideElements/s3fs/blob/master/lib/s3fs.js#L199

**Comment by DavidTPate** [comment]
We actually support passing in Encoding to both [readFile](https://github.com/RiptideElements/s3fs/blob/9504b51462e0bb194da80d8d98f3ee1545e73030/test/file.js#L636-L642) and [writeFile](https://github.com/RiptideElements/s3fs/blob/9504b51462e0bb194da80d8d98f3ee1545e73030/test/file.js#L124-L126) as the parameter in lieu of an options object.

I do see the inconsistency that you are talking about for `readFile` with a promise. The original intention of that was because when reading a file from S3 you might also want to grab the Hash of the file or metadata stored by S3.

**Comment by demetriusnunes** [comment]
Yes, I saw that passing the encoding as an options object would do. It's really no big deal. Just pointed out the inconsistency with "fs" because I'm using your library and "fs" interchangeably, and these things popped out when using "readFile".

**Comment by DavidTPate** [comment]
I definitely understand @demetriusnunes. I'm not trying to push back on it and say the way it works currently is 100% correct. I just need to do some investigation on the impacts of this and it will definitely be a Major version bump since it will fundamentally change how the `readFile` function works.

I'm thinking that a new option will be introduced which will make the current functionality work so that you receive the return from the S3 API instead of the standard FS return. That way it is explicit and we can still support retrieving a file and getting the metadata about it without having to do an extra Head call (minimizing this to a single call to get object metadata is definitely something we use in a number of our projects so they are more performant).
